Firmware/PlatformIO/main_mcu/src/bms.cpp

```diff
@@ -19,6 +27,31 @@ void BMS::Tick(std::chrono::milliseconds elapsed_time)
     // todo
 }
 
+// Find maximum discharge and regen current
+void BMS::GetStateOfEnergy()
+{
+    // Update min voltage (max already set(?))
+    minVoltage = *std::min_element(voltages.begin(), voltages.end());
+
+    float currentPerCell = current[0] / kNumCellsParallel;
+    float invResistancePerCell = kNumCellsParallel / kInternalResistance;
```

Comment:
I would probably use an internalResistancePerSeriesElement = kInternalResistance / kNumCellsParallel and then divide by that later instead of multiplying by invResistancePerCell to make the math more clear